invenio_records_resources/services/files/transfer.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Files transfer."""
```

Comment:
I'm not a 100% convinced about this whole implementeation. It's nice to have it abstracted and with a "factory" to decide the call. However, it feels that the complexity was just _moved to another class_.

Reply:
Overall, I like the complexity/control move since it also helps with extending transfer types easily. What's a bit "obscure" is that this is called from within a component vs. being core logic of the FilesService.